drivers/mysql.go

```diff
@@ -384,94 +470,104 @@ func (db *MySQL) ExecuteDMLStatement(query string) (result string, err error) {
 	return fmt.Sprintf("%d rows affected", rowsAffected), nil
 }
 
-func (db *MySQL) ExecutePendingChanges(changes []models.DbDmlChange, inserts []models.DbInsert) (err error) {
-	queries := make([]string, 0, len(changes)+len(inserts))
-
-	// This will hold grouped changes by their RowId and Table
-	groupedUpdated := make(map[string][]models.DbDmlChange)
-	groupedDeletes := make([]models.DbDmlChange, 0, len(changes))
+func (db *MySQL) ExecutePendingChanges(changes []models.DbDmlChange) (err error) {
+	var query []models.Query
 
-	// Group changes by RowId and Table
 	for _, change := range changes {
-		switch change.Type {
-		case "UPDATE":
-			key := fmt.Sprintf("%s|%s|%s", change.Table, change.PrimaryKeyColumnName, change.PrimaryKeyValue)
-			groupedUpdated[key] = append(groupedUpdated[key], change)
-		case "DELETE":
-			groupedDeletes = append(groupedDeletes, change)
-		}
-	}
-
-	// Combine individual changes to SQL statements
-	for key, changes := range groupedUpdated {
-		columns := []string{}
-
-		// Split key into table and rowId
-		splitted := strings.Split(key, "|")
-		table := db.formatTableName(splitted[0])
-		primaryKeyColumnName := splitted[1]
-		primaryKeyValue := splitted[2]
+		columns := make([]string, len(change.Values))
+		values := make([]interface{}, len(change.Values))
+		valuesPlaceholder := make([]string, len(change.Values))
+
+		for _, cell := range change.Values {
+			valuesPlaceholder = append(valuesPlaceholder, "?")
+
+			switch cell.Type {
+			case models.Empty:
+			case models.Null:
+			case models.String:
+				columns = append(columns, cell.Column)
+			}
 
-		for _, change := range changes {
-			columns = append(columns, fmt.Sprintf("%s='%s'", change.Column, change.Value))
+			switch cell.Type {
+			case models.Empty:
+				values = append(values, "")
+			case models.Null:
+				values = append(values, sql.NullString{})
+			case models.String:
+				values = append(values, cell.Value)
+			case models.Default:
+				break
+			}
 		}
 
-		// Merge all column updates
-		updateClause := strings.Join(columns, ", ")
-
-		query := fmt.Sprintf("UPDATE %s SET %s WHERE %s = '%s';", table, updateClause, primaryKeyColumnName, primaryKeyValue)
-
-		queries = append(queries, query)
-	}
+		switch change.Type {
+		case models.DmlInsertType:
+			queryStr := "INSERT INTO "
+			queryStr += db.formatTableName(change.Database, change.Table)
+			queryStr += fmt.Sprintf(" (%s) VALUES (%s)", strings.Join(columns, ", "), strings.Join(valuesPlaceholder, ", "))
```

Comment:
You could/should a sprintf here to make things easier to read

See https://github.com/jorgerojas26/lazysql/pull/73#discussion_r1685676775